retrofit/src/main/java/retrofit2/KotlinExtensions.kt

```diff
@@ -14,9 +14,84 @@
  * limitations under the License.
  */
 
-// Hide the class from Java consumers.
-@file:JvmName("-KotlinExtensions")
+@file:JvmName("KotlinExtensions")
 
 package retrofit2
 
+import kotlinx.coroutines.suspendCancellableCoroutine
+import kotlin.coroutines.resume
+import kotlin.coroutines.resumeWithException
+
 inline fun <reified T> Retrofit.create(): T = create(T::class.java)
+
+suspend fun <T : Any> Call<T>.await(): T {
+  return suspendCancellableCoroutine { continuation ->
+    continuation.invokeOnCancellation {
+      cancel()
```

Comment:
super cool

Reply:
Should probably test this. Can do in follow-up.